services/m3ctl/main/main.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	etcdclient "github.com/m3db/m3cluster/client/etcd"
+	"github.com/m3db/m3ctl/server"
+	"github.com/m3db/m3ctl/services/m3ctl/serve"
+	"github.com/m3db/m3x/config"
+	"github.com/m3db/m3x/instrument"
+	"github.com/m3db/m3x/log"
+)
+
+type serverConfig struct {
+	// HTTP server listening address.
+	ListenAddress string `yaml:"listenAddress" validate:"nonzero"`
+
+	// HTTP server read timeout.
+	ReadTimeout time.Duration `yaml:"readTimeout"`
+
+	// HTTP server write timeout.
+	WriteTimeout time.Duration `yaml:"writeTimeout"`
+}
+
+// NewHTTPServerOptions create a new set of http server options.
+func (c *serverConfig) NewHTTPServerOptions(
+	instrumentOpts instrument.Options,
+) server.Options {
+	opts := server.NewOptions().SetInstrumentOptions(instrumentOpts)
+	if c.ReadTimeout != 0 {
+		opts = opts.SetReadTimeout(c.ReadTimeout)
+	}
+	if c.WriteTimeout != 0 {
+		opts = opts.SetWriteTimeout(c.WriteTimeout)
+	}
+
+	return opts
+}
+
+type kvClientConfiguration struct {
+	Etcd *etcdclient.Configuration `yaml:"etcd"`
```

Comment:
any reason to define this instead of not using etcdclient.Configuration directly?